docs/guides/upgrading-to-indie-linter-v2.md

```diff
@@ -0,0 +1,77 @@
+# Upgrading to Indie Linter v2
+
+This upgrade guide assumes your package uses Indie Linter API v1 and Message API v1.
+
+## package.json
+
+You need to update the version in the manifest for `linter-indie` service from `1.x.x` to `2.0.0`.
+
+**Before:**
+
+```
+... stuff above
+  "consumedServices": {
+    "linter-indie": {
+      "versions": {
+        "1.0.0": "consumeIndie"
+      }
+    }
+  },
+... stuff below
+```
+
+**After:**
+
+```
+... stuff above
+  "consumedServices": {
+    "linter-indie": {
+      "versions": {
+        "2.0.0": "consumeIndie"
+      }
+    }
+  },
+... stuff below
+```
+
+## index.js
+
+You can restructure your package to use the new `NewIndieDelegate::setMessages` method that is per file, but to keep this guide simple we're gonna upgrade from `OldIndieDelegate::setMessages` to `NewIndieDelegate::setAllMessages` because these two methods are identical.
```

Comment:
`NewIndieDelegate` should be `IndieDelegate` (you are already denoting it's the new one in the word before, no point in using an invalid name.)

Same comment applies to the `OldIndieDelegate`.